### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by szabolcs-keresztes**
- Renaming `fault-counter` to `driving-faults-badge`
- Replacing `TEST_OUTCOME_START_TEST` action with smaller, more accurate actions
- Removing `store$.dispatch` from `journal.effects` and just returning an array of actions instead

### CODE REVIEW

File: src/modules/tests/application-reference/application-reference.actions.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Action } from '@ngrx/store';
+
+export const POPULATE_APPLICATION_REFERENCE = '[JournalEffects] Populate Application Reference';
+
+export class PopulateApplicationReference implements Action {
+  readonly type = POPULATE_APPLICATION_REFERENCE;
+  constructor(public payload: any) {}
```
Comment (Cavalier07) [inline src/modules/tests/application-reference/application-reference.actions.ts]: Could this be typed to Booking instead of any

Comment (johnmorleybjss) [inline src/modules/tests/application-reference/application-reference.actions.ts]: Could you use the `Application` type here?

File: src/modules/tests/candidate/candidate.actions.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Action } from '@ngrx/store';
+
+export const POPULATE_CANDIDATE_DETAILS = '[JournalEffects] Populate Candidate Details';
+
+export class PopulateCandidateDetails implements Action {
+  readonly type = POPULATE_CANDIDATE_DETAILS;
+  constructor(public payload: any) {}
```
Comment (Cavalier07) [inline src/modules/tests/candidate/candidate.actions.ts]: Can this be type Candidate instead of any

File: src/pages/journal/components/test-outcome/test-outcome.ts [hunk 3]
```diff
@@ -2,15 +2,16 @@ import { Component, Input } from '@angular/core';
...
 
 @Component({
   selector: 'test-outcome',
   templateUrl: 'test-outcome.html',
 })
 export class TestOutcomeComponent {
+
   @Input()
-  slot: any;
+  slotId: string;
```
Comment (Cavalier07) [inline src/pages/journal/components/test-outcome/test-outcome.ts]: slotId is a number is the schema

File: src/pages/journal/__tests__/journal.effects.spec.ts [hunk 4]
```diff
@@ -135,13 +135,16 @@ describe('Journal Effects', () => {
...
       false,
       new Date())); // Load in mock journal state
-    spyOn(store$, 'dispatch');
     // ACT
     actions$.next(new journalActions.SelectNextDay());
     // ASSERT
     effects.selectNextDayEffect$.subscribe((result) => {
-      expect(store$.dispatch).toHaveBeenCalledWith(new journalActions.JournalNavigateDay(nextDay));
-      expect(result).toEqual(new journalActions.SetSelectedDate(nextDay));
+      if (result instanceof journalActions.SetSelectedDate)  {
```
Comment (johnmorleybjss) [inline src/pages/journal/__tests__/journal.effects.spec.ts]: Conditions around assertions seem a bit fishy, why are these required?

File: src/pages/journal/components/test-outcome/test-outcome.ts (hunk 3 above)
Comment (szabolcs-keresztes) [reply src/pages/journal/components/test-outcome/test-outcome.ts]: Fair point.
I'm not a big fan of having `slotId` as a number, but will change it for consistency.

File: src/modules/tests/application-reference/application-reference.actions.ts (hunk 1 above)
Comment (szabolcs-keresztes) [reply src/modules/tests/application-reference/application-reference.actions.ts]: Will make it Application

File: src/modules/tests/candidate/candidate.actions.ts (hunk 2 above)
Comment (szabolcs-keresztes) [reply src/modules/tests/candidate/candidate.actions.ts]: cool

File: src/pages/journal/__tests__/journal.effects.spec.ts (hunk 4 above)
Comment (ammarhaiderbjss) [inline src/pages/journal/__tests__/journal.effects.spec.ts]: do we need these ifs?

Comment (szabolcs-keresztes) [reply src/pages/journal/__tests__/journal.effects.spec.ts]: I feel that this is a hackie solution. But couldn't do it better. So the function in subscribe will be called twice because of the elements of the action array will be arrive one by one. First time the first if will be true so we can make the right assertion and second time the second.

Let's have a think about this in the future and come up with a better solution :)

File: src/modules/tests/application-reference/application-reference.actions.ts (hunk 1 above)
Comment (szabolcs-keresztes) [reply src/modules/tests/application-reference/application-reference.actions.ts]: just pushed the update

File: src/pages/journal/__tests__/journal.effects.spec.ts (hunk 4 above)
Comment (szabolcs-keresztes) [reply src/pages/journal/__tests__/journal.effects.spec.ts]: Agree. See my answer below
